[FIX] l10n_in_edi: remove state validation for overseas partner

hamo-odoo opened the pull request:
This commits removes the state validation for `l10n_in_edi` for overseas partner and now state will be only required for e-invoicing for partner having country `India`

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

robodoo commented:
[Pull request status dashboard](https://mergebot.odoo.com/odoo/odoo/pull/154118).

hamo-odoo commented:
@clbr-odoo Can you please review this PR?

clbr-odoo commented:
Hello @hamo-odoo it looks like more of a `[FIX]` than an `[IMP]` ?

clbr-odoo reviewed (commented):
Hello @hamo-odoo it looks like more of a `[FIX]` than an `[IMP]` ?
If it's really an `[IMP]` to you is there a task-id related to it ?

Also there is a small typo in your commit title "valdiation" :wink:

hamo-odoo commented:
> Hello @hamo-odoo it looks like more of a `[FIX]` than an `[IMP]` ? If it's really an `[IMP]` to you is there a task-id related to it ?
> 
> Also there is a small typo in your commit title "valdiation" 😉

@clbr-odoo I fixed and updated my typo's as much as you did in your comment :wink:

csnauwaert commented:
@robodoo r+
